server/expense_tag.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"errors"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+)
+
+var expenseTagTable = `
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+CREATE TABLE IF NOT EXISTS expense_tag (
+	expense_id uuid,
+	tag_id uuid,
+	created_at timestamp with time zone
+)`
+
+type ExpenseTag struct {
+	ExpenseID string    `db:"expense_id" json:"expense_id"`
+	TagID     string    `db:"tag_id" json:"tag_id"`
+	CreatedAt time.Time `db:"created_at" json:"created_at"`
+}
+
+type ExpenseTagOutput struct {
+	TagID   string `db:"id" json:"tag_id"`
+	TagName string `db:"name" json:"tag_name"`
+}
+
+func insertTagsOfExpense(c *gin.Context, tagIDs *[]string, expenseID string) (error, int) {
+	err, errCode := checkIfTagsExist(c, tagIDs)
+	if err != nil {
+		return err, errCode
+	}
+
+	db := GetDB()
+	/* only start inserting data if tags exist */
+	for _, tagID := range *tagIDs {
+		_, err = db.Exec("INSERT INTO expense_tag VALUES ($1, $2, now())", expenseID, tagID)
+
+		/* if there is a database error */
+		if err != nil {
+			return err, 500
+		}
+	}
+
+	return nil, 200
+}
+
+func checkIfTagsExist(c *gin.Context, tagIDs *[]string) (error, int) {
+	tags := []Tag{}
+	userID := c.MustGet("userID")
+	var err error
+
+	for _, tagID := range *tagIDs {
+		/* check if tag ID is a uuid */
+		if _, err = uuid.Parse(tagID); err != nil {
+			return err, 400
+		}
+	}
+
+	db := GetDB()
+
+	query, args, err := sqlx.In("SELECT * FROM tag WHERE user_id=(?) and id IN (?) ", userID, *tagIDs)
```

Comment:
Are you sure the first () is necessary / works?

Reply:
You wanted it to be save :smile: 
Does it make any difference regarding performance?